sys/shell/src/shell.c

```diff
@@ -21,543 +21,432 @@
-#include "crc/crc16.h"
+#include "console/console.h"
+#include "sysinit/sysinit.h"
 #include "shell/shell.h"
-#include "shell_priv.h"
-
-static shell_nlip_input_func_t g_shell_nlip_in_func;
-static void *g_shell_nlip_in_arg;
-
-static struct os_mqueue g_shell_nlip_mq;
-
-#define SHELL_HELP_PER_LINE     6
-#define SHELL_MAX_ARGS          20
-
-static int shell_echo_cmd(int argc, char **argv);
-static int shell_help_cmd(int argc, char **argv);
-int shell_prompt_cmd(int argc, char **argv);
-int shell_ticks_cmd(int argc, char **argv);
-
-
-static void shell_event_console_rdy(struct os_event *ev);
-
-/* Shared queue that the shell uses for work items. */
-static struct os_eventq *shell_evq;
-
-static struct shell_cmd g_shell_echo_cmd = {
-    .sc_cmd = "echo",
-    .sc_cmd_func = shell_echo_cmd
-};
-static struct shell_cmd g_shell_help_cmd = {
-    .sc_cmd = "?",
-    .sc_cmd_func = shell_help_cmd
-};
-static struct shell_cmd g_shell_prompt_cmd = {
-   .sc_cmd = "prompt",
-   .sc_cmd_func = shell_prompt_cmd
-}; 
-static struct shell_cmd g_shell_ticks_cmd = {
-   .sc_cmd = "ticks",
-   .sc_cmd_func = shell_ticks_cmd
-};
-static struct shell_cmd g_shell_os_tasks_display_cmd = {
-    .sc_cmd = "tasks",
-    .sc_cmd_func = shell_os_tasks_display_cmd
-};
-static struct shell_cmd g_shell_os_mpool_display_cmd = {
-    .sc_cmd = "mempools",
-    .sc_cmd_func = shell_os_mpool_display_cmd
-};
-static struct shell_cmd g_shell_os_date_cmd = {
-    .sc_cmd = "date",
-    .sc_cmd_func = shell_os_date_cmd
-};
-
-static struct os_event shell_console_rdy_ev = {
-    .ev_cb = shell_event_console_rdy,
-};
-
-static struct os_mutex g_shell_cmd_list_lock;
-
-static char *shell_line;
-static int shell_line_len;
-static char *argv[SHELL_MAX_ARGS];
-
-static STAILQ_HEAD(, shell_cmd) g_shell_cmd_list =
-    STAILQ_HEAD_INITIALIZER(g_shell_cmd_list);
-
-static struct os_mbuf *g_nlip_mbuf;
-static uint16_t g_nlip_expected_len;
-
-static struct os_eventq *
-shell_evq_get(void)
-{
-    return shell_evq;
-}
 
-void
-shell_evq_set(struct os_eventq *evq)
-{
-    os_eventq_designate(&shell_evq, evq, NULL);
-}
+#define SHELL_PROMPT "shell> "
 
-int
-shell_cmd_list_lock(void)
+#define MODULE_NAME_MAX_LEN     20
+
+/* additional chars are "> " (include '\0' )*/
+#define PROMPT_SUFFIX 3
+#define PROMPT_MAX_LEN (MODULE_NAME_MAX_LEN + PROMPT_SUFFIX)
+
+static struct shell_module shell_modules[MYNEWT_VAL(SHELL_MAX_MODULES)];
+static size_t num_of_shell_entities;
+
+static const char *prompt;
+static char default_module_prompt[PROMPT_MAX_LEN];
+static int default_module = -1;
+
+static shell_cmd_func_t app_cmd_handler;
+static shell_prompt_function_t app_prompt_handler;
+
+static struct console_input buf[MYNEWT_VAL(SHELL_MAX_CMD_QUEUED)];
+
+static struct os_eventq avail_queue;
+static struct os_event shell_console_ev[MYNEWT_VAL(SHELL_MAX_CMD_QUEUED)];
+
+static const char *
+get_prompt(void)
 {
-    int rc;
+    if (app_prompt_handler) {
+        const char *str;
```

Comment:
coding style:  variables should be declared add beginning of function, not block